### ISSUE TITLE
Clear all conversation history

### ISSUE THREAD

**Issue by Sb09897** [microsoft/vscode-copilot-release#209]
Hi All,

Like in ChatGPT we need the clear conversations button on the side panel.

What do you thing about m suggestion?

Let me know here ASAP

Regards,
Sb09897

**Comment by roblourens** [comment]
You can also open the history picker and press the X to delete individual entries. Let me know if you are asking for something else.

**Comment by Sb09897** [comment]
> @Sb09897 Since there is a clear button, I assume you mean cleaning the whole history?
> 
> <img alt="image" width="452" src="https://user-images.githubusercontent.com/8599/243445731-2978a61a-fb83-4f0d-99ca-88659786b342.png">

Yes like in ChatGPT there is a button to clear all prevous message lists.

We should have one here also so that developers can have ease of access to chats rather than cluttering up the whole list of conversations on searching.

@roblourens by using the x shortcut we can delete individual entries i am talking about clearing them all at once.
![image](https://github.com/microsoft/vscode-copilot-release/assets/102576743/229cbe58-562b-44e8-b2cf-e321f9ca131e)
Basically i am taking about the clear conversations button present in ChatGPT

**Comment by Sb09897** [comment]
thanks @roblourens for understanding me in the right way!

**Comment by digitarald** [comment]
@Sb09897 👍 on feature requests is one of the signals, but more is outlined in our [issue tracking wiki](https://github.com/microsoft/vscode/wiki/Issue-Tracking).

**Comment by Sb09897** [comment]
> @Sb09897 👍 on feature requests is one of the signals, but more is outlined in our [issue tracking wiki](https://github.com/microsoft/vscode/wiki/Issue-Tracking).

Okay!

**Comment by sovetski** [comment]
> @Sb09897 Since there is a clear button, I assume you mean cleaning the whole history?
> 
> <img alt="image" width="452" src="https://user-images.githubusercontent.com/8599/243445731-2978a61a-fb83-4f0d-99ca-88659786b342.png">

It is difficult to understand that this icon is made to clear history, replacing it by another icon and changing its position will be nice I think

**Comment by Sb09897** [comment]
> > @Sb09897 Since there is a clear button, I assume you mean cleaning the whole history?
> > <img alt="image" width="452" src="https://user-images.githubusercontent.com/8599/243445731-2978a61a-fb83-4f0d-99ca-88659786b342.png">
> 
> It is difficult to understand that this icon is made to clear history, replacing it by another icon and changing its position will be nice I think

I agree!

- [ ] @digitarald @roblourens add this to your task

**Comment by barry-hall** [comment]
> You can also open the history picker and press the X to delete individual entries. Let me know if you are asking for something else.

Is there a keybinding / when condition I can add to remove an item with a keybinding instead of needing to use the mouse to hit the X?

**Comment by Sb09897** [comment]
> > You can also open the history picker and press the X to delete individual entries. Let me know if you are asking for something else.
> 
> Is there a keybinding / when condition I can add to remove an item with a keybinding instead of needing to use the mouse to hit the X?

No that is still not present that the reason why i created this issue.

**Comment by Sb09897** [comment]
Thank you @roblourens!

**Comment by ZeyadHabash** [comment]
I cannot seem to find the clear all workspace chats command anywhere

**Comment by branden1992** [comment]
Recover all messages please sorry

**Comment by PilotGFX** [comment]
![image](https://github.com/microsoft/vscode-copilot-release/assets/121142153/adfe493a-f129-4ef0-84f6-02e370e1256c)

The delete button has vanished from my UI and i am forced to delete all interactions message by message which is tedious. How to fix it?

**Comment by roblourens** [comment]
The + button does the same thing- start a new chat

**Comment by PilotGFX** [comment]
Oh... seems kind of obvious now, thanks roblourens!
